File: frontend-server/src/app/model/dto/shift-pattern-dto.ts
import { ShiftPattern } from "../shift-pattern";
import { PatternUnit } from "../pattern-unit";
import { ShiftPatternGenerationRule } from "../shift-pattern-generation-rule";
import { BasicDTO } from "./basic-dto";

export class ShiftPatternDTO extends BasicDTO<ShiftPattern, PatternUnit> {
  generationRules: ShiftPatternGenerationRule[];
}

File: core/src/main/java/com/averude/uksatse/scheduler/core/dto/ShiftPatternDTO.java
package com.averude.uksatse.scheduler.core.dto;

import com.averude.uksatse.scheduler.core.entity.PatternUnit;
import com.averude.uksatse.scheduler.core.entity.ShiftPattern;
import com.averude.uksatse.scheduler.core.entity.ShiftPatternGenerationRule;
import lombok.Getter;
import lombok.Setter;

import java.util.List;

@Getter
@Setter
public class ShiftPatternDTO extends BasicDto<ShiftPattern, PatternUnit> {
    private List<ShiftPatternGenerationRule> generationRules;
}
